interview-questions.md

```diff
@@ -9,59 +9,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: THIS is a method to identify objects and attatch or tie them to a function within React.
 
-  Researched answer:
+  Researched answer:THIS basically gives direction to the object it is attached too.  It basically tells you were to get data or value from.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a development tool that lets devopers see their work and any changers to their work in real time on the web.  What makes React so unique is it stores all the necessary Javascript library codes and has them ready for the developer to use when needed. 
 
-  Researched answer:
+  Researched answer:React is a front end framework that works with virtual DOM (Document Object Model) and created by Faccebook.  Basically a library of Javascript that is scalable and updates in real time.
+  React is flexible, reliable, easy to use, support driven, has a large user communkity and its easy to test.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: State is a required lifecycle method in a React class component becuase to ties various functions together so the desired outcome is a reality.
 
-  Researched answer:
+  Researched answer:The render() method is the only method required in a class component.
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is the merging of HTML and JAVASCRIPT code using the curly brackets and / .  So in a .JS file we can have both HTML and JAVASCRIPT working together on the same code line.
+  HTML is showing the basic foundation of the webpage while the JSX is taking that foundation and giving it functionality.  
 
   Researched answer:
-
+JSX is an extension of javascript that lets you write html code within javascript.  One sntax difference is tags self close in JSX.  So in 
+<div></div> in HTML is <div /> in JSX.
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Yarn is what ties all the code together in the React application.  All the files in the SRC folder are modified when running yarn.
```

Comment:
Yarn does not really manage the SRC folder, however it does manage all of the node modules.